src/lib/mdns/Resolver.h

```diff
@@ -40,12 +40,15 @@ struct CommissionableNodeData
     // TODO(cecille): is 4 OK? IPv6 LL, GUA, ULA, IPv4?
     static constexpr int kMaxIPAddresses = 5;
     // Largest host name is 64-bits in hex.
-    static constexpr int kHostNameSize = 16;
+    static constexpr int kHostNameSize   = 16;
+    static constexpr int kDeviceNameSize = 32;
 
     char hostName[kHostNameSize + 1];
     uint16_t longDiscriminator;
     uint16_t vendorId;
     uint16_t productId;
+    uint16_t deviceType;
```

Comment:
Warning: this isn't wrong, but it's going to conflict with https://github.com/project-chip/connectedhomeip/pull/7006.